### PR TITLE
Wrecked Ship: Use Flash Suit

### PR DESCRIPTION
**Pull request by kjbranch**

### CODE REVIEW

File: region/wreckedship/main/Bowling Alley.json [hunk 1]
```diff
@@ -1003,10 +1014,55 @@
...
+        {"or": [
+          {"shinespark": {"frames": 32, "excessFrames": 3}},
+          {"and": [
+            "HiJump",
+            {"shinespark": {"frames": 29, "excessFrames": 3}}
+          ]},
+          {"and": [
+            {"shinespark": {"frames": 5, "excessFrames": 3}},
+            {"or": [
+              "canWalljump",
```
Comment (osse101) [inline region/wreckedship/main/Bowling Alley.json:1057]: harder-than-precise walljump (and consecutive) or phantoon not defeated.
Or if you can flash suit, you can also manage this?

File: region/wreckedship/main/Bowling Alley.json [hunk 2]
```diff
@@ -1003,10 +1014,55 @@
...
-        {"shinespark": {"frames": 40}}
+        {"or": [
+          {"shinespark": {"frames": 40, "excessFrames": 3}},
+          {"and": [
+            "canShinechargeMovement",
+            "canMidairShinespark",
+            {"shinespark": {"frames": 32, "excessFrames": 2}}
+          ]},
+          {"and": [
+            "canShinechargeMovementComplex",
```
Comment (osse101) [inline region/wreckedship/main/Bowling Alley.json:1025]: With wall jumps my best was 10 shinespark frames but getting past the spikes was extremely annoying.
With hijump I made it to the top before sparking.

File: region/wreckedship/main/Wrecked Ship East Super Room.json [hunk 3]
```diff
@@ -523,6 +523,21 @@
...
+      "name": "WS East Supers Robot Clip, Use Flash Suit",
+      "requires": [
+        "canMidAirMorph",
+        "canKago",
+        "canHorizontalShinespark",
+        {"useFlashSuit": {}},
+        {"shinespark": {"frames": 1, "excessFrames": 1}}
+      ],
+      "note": [
+        "Kago into the worker robot and shinespark horizontally into the wall in order to break the bomb block with minimal Energy loss."
```
Comment (osse101) [inline region/wreckedship/main/Wrecked Ship East Super Room.json:537]: Interesting that this has to be a lowish horizontal shinespark -- not from a crouch jump.  Or the robot pushes you up too high.

File: region/wreckedship/main/Bowling Alley.json (hunk 1 above)
Comment (kjbranch) [reply region/wreckedship/main/Bowling Alley.json:1081]: I think with flash suit this should be fine.

**Pull request merged**